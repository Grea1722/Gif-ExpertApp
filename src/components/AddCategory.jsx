import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('')

    const onInputChange = (event) =>{
        setinputValue(event.target.value)
    }

    const onSubmit = (event) => {
      event.preventDefault();
      if(inputValue.trim().length <= 1 ) return;
      // setCategories(cats=>([inputValue,...cats]));
      onNewCategory(inputValue.trim());
      setinputValue('');
    }
  return (
    <form onSubmit={onSubmit}>
      <input type='text' 
      placeholder='Buscar Gifs' onChange={onInputChange}
      value={inputValue}
       />
    </form>
  )
}