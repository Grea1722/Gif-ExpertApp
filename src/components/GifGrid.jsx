import { useFecthGifs } from "../hooks/useFecthGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const {images, isLoading}=useFecthGifs(category)
  // const [images, setimages] = useState([]);
  // useEffect(() => {
  //   //No se debe usar async en useEffect
  //   GetGifs(category).then((newImages) => setimages(newImages));
  // }, []);

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && <h2>Cargando!!!</h2>
      }
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} 
          {...image}/>
        ))}
      </div>
    </>
  );
};

