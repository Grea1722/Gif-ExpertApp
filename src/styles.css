* {
    font-family: Helvetica, Arial, sans-serif;
    text-align: center;
  }
  
  body {
    margin: 0%;
  }
  
  input {
    background-color: white;
    border-radius: 5px;
    border: none;
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 80%;
    text-transform: uppercase;
    box-shadow: 0px 0px 15px rgba(0, 0, 0, 0.443);
  }
  h1{
    background-color: red;
    width: 30vw;
    margin: auto;
    margin-bottom: 10px;
    padding: 20px;
    background-color: white;
    border-radius: 10px;
    box-shadow: 0px 10px 10px rgba(0, 0, 0, 0.443);
  }

  .form{
    background-color: #381cf1;
    height: 40px;
    margin-bottom: 12vh;
    border-radius: 0px 0px 0px 60px;
    padding-top: 7vh;
  }
  
  h2 {
    font-size: 1.5rem;
  }
  
  h3 {
    font-size: 3rem;
    margin-bottom: 5px;
    text-transform: uppercase;
  }
  
  .card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-evenly;
    padding: 0 40px;
  }
  
  .card {
    align-content: center;
    align-items: center;
    background-color: white;
    border-radius: 10px;
    box-shadow: 10px 5px 10px rgba(0, 0, 0, 0.443);
    display: flex;
    flex-direction: column;
    height: 0%;  
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
    width: 30%;
  }
  
  .card p {
    background-color: white;
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 5px 20px 0px 20px;
    text-align: center;
  }
  
  .card img {
    width: 100%;
  }
