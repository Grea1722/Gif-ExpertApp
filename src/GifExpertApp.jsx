import React, { useState } from 'react'
import { AddCategory,GifGrid } from './components';

export const GifExpertApp = () => {
  const [categories, setcategories] = useState(['Dragon Ball']);

  const onAddCategory = (newCategory) => {
    if(categories.includes(newCategory)) return;
    setcategories([newCategory,...categories]);
  }

  return (
    <>
    <div className='form'>
      <h1>Gif Expert App</h1>
    </div>
    
    <AddCategory 
    // setCategories={setcategories}
    onNewCategory={(value)=>onAddCategory(value)}
    />
    {
      categories.map((category) => 
        (<GifGrid key = {category} category={category}/>)
      )
    }
    </>
  )
}
