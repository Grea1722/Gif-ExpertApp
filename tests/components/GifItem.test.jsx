import {render,screen } from "@testing-library/react";
import { GifItem } from "../../src/components";


describe('Prueba en <GifItem />', () => { 
    const title = 'Goku'
    const url = 'https://dragon-ball.com/goku.jpg'

    test('Debe hacer match con snapshot',()=>{
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el URL y ALT indicado',() => {
        render(<GifItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getAllByRole('img').src).toBe(url);
        // expect(screen.getAllByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url )
        expect( alt ).toBe( title );
    })

    test('debe mostrar el titulo del componente', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText( title )).toBeTruthy();
    })
 });