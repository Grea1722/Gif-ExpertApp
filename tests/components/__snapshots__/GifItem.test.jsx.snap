// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem /> Debe hacer match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://dragon-ball.com/goku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
